compair/static/modules/student_view/student-view-module.js

```diff
@@ -0,0 +1,48 @@
+// Provides the services and controllers for Student View
+
+(function() {
+
+var module = angular.module('ubc.ctlt.compair.studentview',
+    [
+        'ngResource',
+        'ubc.ctlt.compair.common.form',
+        'ubc.ctlt.compair.course',
+        'ubc.ctlt.compair.toaster'
+    ]
+);
+
+module.controller(
+    "StudentViewController",
+    [ "$scope", "UserResource", "ClassListResource", "Session", "$uibModalInstance",
+    function($scope, UserResource, ClassListResource, Session, $uibModalInstance) {
+        $scope.studentView = $scope.studentView || {};
+
+        $scope.submitted = false;
+        $scope.studentView.courseId = $scope.currentCourseId;
+        $scope.studentView.userId = null;
+        UserResource.getTeachingUserCourses().$promise.then(function(ret) {
+            $scope.courses = ret.courses;
+        });
+        $scope.getClassList = function() {
+            if ($scope.studentView.courseId) {
+                ClassListResource.get({'courseId': $scope.studentView.courseId}).$promise.then(function (ret) {
+                    $scope.classlist = ret.objects;
+                });
+            }
+        }
+        $scope.getClassList();
+
+        $scope.studentViewSubmit = function() {
+            $scope.submitted = true;
+
+            var studentView = $scope.studentView || {};
+            Session.impersonate(studentView.courseId, studentView.userId);
```

Comment:
Changed `Session.impersonate` to a promise and keep user on the modal dialog if there is an error.

Actually this can be an embarrassing situation.  If `ImpersonateResource.impersonate` succeed but `session_exports.refresh` failed, user will be kind of stuck.  Backend will reject any POST or DELETE as impersonation has started.  However frontend is still showing the instructor view.

Will also add logic in `Session.impersonate` to destroy the session before `session_exports.refresh`.  So that at least a successful browser reload will refresh everything.

```
                return ImpersonateResource.impersonate({'userId': userId}).$promise.then(function(ret) {
                    // Destroy the session first. In case the refresh failed (e.g. network issue), 
                    // a success reload of the page will show the correct student view.
                    session_exports.destroy();
                    return session_exports.refresh(false).then(function () {
```